### ISSUE TITLE
3v3 pin is 5v on a BX from Crowd Supply

### ISSUE THREAD

**Issue by richgel999** [tinyfpga/TinyFPGA-BX#39]
I first had to work around the "no metadata" issue (by modifying a python file, as others describe). I'm now able to program the FPGA using the default template and I see the boot LED blink "SOS". 

I noticed that both LED's are extremely (too) bright, which is odd. They've always been this way since when I first plugged it in.

I measured the 3.3v pin (pin #25), and it's measuring 5v when plugged into USB. If I unplug USB and supply 4.5v to the "in" pin, I get 4.5v at the 3.3v pin. The 3.3v regulator doesn't appear to be working. 

I'm at a loss at what to do now. Perhaps try another platform.

**Comment by jeffpc** [comment]
I've gotten two BX boards from Mouser recently and they both fail to finish the USB handshake (https://github.com/tinyfpga/TinyFPGA-Bootloader/issues/74). Additionally, they both have 5V on the 3.3V rail.  There are *no* shorts between the two rails; the 1.2V rail is spot on.  So, I also suspect that the MIC5504-3.3YM5 voltage regulator is bad.

**Comment by leaningktower** [comment]
Hi jeffpc, I found the same issue as you mentioned recently and I also can not find a short. I found this problem because I want to use an external clock input and it turns out I need at least 5 V to make the it work and it turns out the 3.3 V is totally wrong.
After simple testing, I am also pretty sure the LDO is not broken but it is a different LDO rather than the MIC5501 in the schematic. The code on the 3.3 V LDO is g5k, and it turns out to be AP2210K-3.3. However, the pin 4 of this LDO is not NC pin and can not be connected together with the VOUT pin 5. And that is why the LDO gives an saturated output that looks like a short. The simple solution I did is cut the pin 4 and 3.3 V is back.

**Comment by jeffpc** [comment]
Good catch! Both of my boards have a 3.3V regulators marked G5K.  I cut pin 4 on one and now the 3.3V rail is 3.3V like it should be. Sadly, this didn't fix my other problem (USB handshake).

**Comment by leaningktower** [comment]
I purchased my board through crowd supply and waited it for at least 6 months. I am also suffered from handshake issue and use the dirty method mentioned by one of the post here.  I guess all the broad shipped middle this year suffered from the same issue. Unfortunately I mistakenly broke the FPGA today, instead of waiting for the crazy lead time, I decided to purchase several iCE40LP8K chips and BGA soldering tools.

**Comment by jonathancollins514** [comment]
datasheet:
https://www.diodes.com/assets/Datasheets/AP2210.pdf

pin4:
![pin4](https://github.com/tinyfpga/TinyFPGA-BX/assets/155577920/bd6f1854-aabe-42d6-a076-6744bed62c5c)

![pin4 pic](https://github.com/tinyfpga/TinyFPGA-BX/assets/155577920/04f19d48-925e-481e-9854-b3d0ecdb6f98)

**Comment by arndgo** [comment]
Thanks, works. Getting 3.3V now.
Great work.

**Comment by eyz** [comment]
> I cut pin 4 on one and now the 3.3V rail is 3.3V like it should be. 

@jeffpc What did you do? I can't tell exactly from the picture. Did you simply snip off pin 4 at the AP2210? Do you know why that works?

**Comment by eyz** [comment]
I snipped pin 4 with an angle cutter on the G5K 3.3v regulator side of the lead downward from above, and confirmed that the whole lead down to the PCB pad was removed successfully. Getting 3.3v now 🎉 as others have stated here -

<img width="374" alt="image" src="https://github.com/user-attachments/assets/534f3c2b-ce40-4091-83f9-5289519dab9d">

The G5K and pad on the PCB are quite separate in height, and nowhere near connected now -
<img width="861" alt="image" src="https://github.com/user-attachments/assets/4e9b1700-2d1e-4229-b9a4-aea5fb40b9ac">
<img width="433" alt="image" src="https://github.com/user-attachments/assets/b62b6f18-24e1-414a-a46f-628aa464d2e6">

**Comment by eyz** [comment]
Let's try to ensure the next round of boards won't have 5V on the 3.3V pin (https://github.com/tinyfpga/TinyFPGA-BX/issues/39#issuecomment-2386551707) due to a `AP2210K-3.3` part swap having pin 4 wired, and hasn't written zeros to the area of the PROM on the FPGA where metadata needs to exist to allow `tinyprog` to identify the board (https://github.com/tinyfpga/TinyFPGA-BX/issues/37#issuecomment-1694056732).

Per Mouser, there is a batch expected out around May 2025. If these are also botched, let's try and get PCBA files to produce these instead.

If anyone has a contact for who is producing the next run, could you please reach out to them and let them know of these two critical issues? A Mouser contact here who is on Github would also be appreciated to share the findings.
